pkg/user/parallel_tx_submission.go

```diff
@@ -0,0 +1,505 @@
+package user
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"sync/atomic"
+
+	"cosmossdk.io/x/feegrant"
+	"github.com/celestiaorg/celestia-app/v6/pkg/appconsts"
+	"github.com/celestiaorg/go-square/v3/share"
+	"github.com/cosmos/cosmos-sdk/crypto/hd"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	bank "github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+// SubmissionJob represents a transaction submission task for parallel processing
+type SubmissionJob struct {
+	Blobs    []*share.Blob
+	Options  []TxOption
+	Ctx      context.Context
+	ResultsC chan SubmissionResult
+}
+
+// SubmissionResult contains the result of a parallel transaction submission
+type SubmissionResult struct {
+	Signer     string
+	TxResponse *TxResponse
+	Error      error
+}
+
+// txQueue manages parallel transaction submission
+type txQueue struct {
+	client      *TxClient
+	jobQueue    chan *SubmissionJob
+	workers     []*txWorker
+	started     atomic.Bool
+	ctx         context.Context
+	cancel      context.CancelFunc
+	initialized atomic.Bool // whether workers have been initialized
+	wg          sync.WaitGroup
+}
+
+// txWorker represents a worker that processes transactions using a specific account
+type txWorker struct {
+	id          int
+	accountName string
+	address     string
+	client      *TxClient
+	jobQueue    chan *SubmissionJob
+}
+
+const (
+	defaultParallelQueueSize = 100
+)
+
+func newTxQueue(client *TxClient, numWorkers int) *txQueue {
+	pool := &txQueue{
+		client:   client,
+		jobQueue: make(chan *SubmissionJob, defaultParallelQueueSize),
```

Comment:
[nit] it looks like `defaultParallelQueueSize` can't be overridden so IMO we could remove the `default` from the name.

In other words, rename: `defaultParallelQueueSize` to `parallelQueueSize`